### ISSUE TITLE
Job summary sometimes misses task groups, leading to plan failures

### ISSUE THREAD

**Issue by ExalDraen** [hashicorp/nomad#5939]
### Nomad version
Nomad v0.9.1 (4b2bdbd9ab68a27b10c2ee781cceaaf62e114399)

### Operating system and Environment details
Ubuntu 16.04, running on GCP
```
$ uname -a
Linux nomad-consul-servers-dev-4mf7 4.15.0-1032-gcp #34~16.04.1-Ubuntu SMP Wed May 8 13:26:26 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```
### Issue

The `JobSummary` struct of a job sometimes omits the TaskGroup summary for the task groups contained within the job. This leads to failures when trying to `plan` an update to the job. 

The error we observe is this:
```
[2019-07-05T07:59:32.159Z] 2019-07-05T15:59:32+08:00 |ERRO| levant/plan: error when running plan error="Unexpected response code: 500 (error updating job summary: unable to find task group in the job summary: algo-start-grp)"
```
NB: we use [levant](https://github.com/jrasell/levant) to help with deployments, but this just calls the nomad API. 

I have verified with the nomad api that indeed the job summary for that job is indeed empty:
```
curl ${NOMAD_ADDR}/v1/job/prod-algo-start-jnid/summary
{"Children":{"Dead":5,"Pending":0,"Running":0},"CreateIndex":8820,"JobID":"prod-algo-start-jnid","ModifyIndex":64722,"Namespace":"default","Summary":{}}
```
However, querying just the "Read Job" API for that same job shows that there _are_ in fact, task groups:

```
curl ${NOMAD_ADDR}/v1/job/prod-algo-start-jnid
{
   (...)
  "TaskGroups": [
    {
      "Affinities": null,
      "Constraints": null,
      "Count": 1,
      "EphemeralDisk": {
        "Migrate": false,
        "SizeMB": 300,
        "Sticky": false
      },
      "Meta": null,
      "Migrate": null,
      "Name": "algo-start-grp",
    (...)
```

Please can you help me troubleshoot why the `JobSummary` for certain job is sometimes missing the task groups. 

### Reproduction steps
I do not currently have a solid sets of steps to reproduce this 100%, but we have noticed the following:

* this only occurs for batch jobs with a `cron` expression
* Changing the _name_ of the task group solves the issue in so far as `plan` (and deploy) succeeds

### Job file (if appropriate)
We template our job files using https://github.com/jrasell/levant, but I have attached an example rendered job file for a job that exhibits the problem:
[prod-algo-start-jnid.nomad.txt](https://github.com/hashicorp/nomad/files/3366629/prod-algo-start-jnid.nomad.txt)

Note that I have elided a few things with (...)

**Comment by preetapan** [comment]
@ExalDraen Job summaries are updated when the job is updated or any allocations are updated. 

In your jobs you are looking at the summary of the parent job ID, which will not contain task group level summaries for its children jobs. So its expected for Summary to be empty for the parent job ID. 

If you know the job IDs of the children jobs, those should have their summaries populated correctly.

**Comment by ExalDraen** [comment]
@preetapan thank you for your feedback, but I don't quite follow your comment. Additionally, I think that there even if what you said is accurate, there is still an issue. Please see below.

----

You said that parent job IDs will not contain task group level summaries for its children jobs. However, I can see several examples on my cluster that _do_ show task group level summaries for batch type parent jobs.

For example, here is the summary for a similar batch (+cron) parent job
```
$ curl ${NOMAD_ADDR}/v1/job/dev-ahab-XCME-start/summary
{"Children":{"Dead":13,"Pending":0,"Running":0},"CreateIndex":54714,"JobID":"dev-ahab-XCME-start","ModifyIndex":67825,"Namespace":"default","Summary":{"ahab-start":{"Complete":0,"Failed":0,"Lost":0,"Queued":0,"Running":0,"Starting":0},"ahab":{"Complete":0,"Failed":0,"Lost":0,"Queued":0,"Running":0,"Starting":0}}}
```

----

In addition, even leaving aside what you said, I still believe there is an issue. 

I think you may have missed the fact that we hit this error when using the [plan endpoint of the Nomad API](https://www.nomadproject.io/api/jobs.html#create-job-plan) - there's actually nothing in our process / code that cares about summaries, this is just a part of what happens when Nomad runs plan (as I understand it). 

To put it another way: the plan API endpoint sometimes fails for us on these type of batch (+cron) jobs.
